ios/src/TGMapViewController.mm

```diff
@@ -23,26 +25,116 @@ @interface TGMapViewController ()
 
 @implementation TGMapViewController
 
-- (void)setPosition:(TangramGeoPoint)position {
+- (void)loadSceneFile:(NSString*)path {
+    if (!self.map) {
+        return;
+    }
+
+    self.scenePath = path;
+    self.map->loadScene([path UTF8String]);
+    self.renderRequested = YES;
+}
+
+- (void)loadSceneFileAsync:(NSString*)path {
+    if (!self.map) { return; }
+
+    self.scenePath = path;
+
+    MapReady onReadyCallback = [self, path](void* _userPtr) -> void {
+        if (self.mapViewDelegate && [self.mapViewDelegate respondsToSelector:@selector(mapView:didLoadSceneAsync:)]) {
+            [self.mapViewDelegate mapView:self didLoadSceneAsync:path];
+        }
+
+        self.renderRequested = YES;
+    };
+
+    self.map->loadSceneAsync([path UTF8String], false, onReadyCallback, nullptr);
+}
+
+- (void)queueSceneUpdate:(NSString*)componentPath withValue:(NSString*)value {
+    if (!self.map) { return; }
+
+    self.map->queueSceneUpdate([componentPath UTF8String], [componentPath UTF8String]);
+}
+
+- (void)requestRender {
+    if (!self.map) { return; }
+
+    self.renderRequested = YES;
+}
+
+- (CGPoint)lngLatToScreenPosition:(TGGeoPoint)lngLat {
+    static const CGPoint nullCGPoint = {(CGFloat)NAN, (CGFloat)NAN};
```

Comment:
Returning `nil` would mean either returning an objective-c object or a pointer to a `CGPoint` (CGPoint is a c-type struct). I also prefer the `nil` option but here `NaN` seems to be the only alternative if we want to keep on using `CGPoint`.